pkg/canary/comparator/comparator_test.go

```diff
@@ -218,6 +218,33 @@ func TestEntryNeverReceived(t *testing.T) {
 
 }
 
+// Ensure that if confirmMissing calls pile up and run concurrently, this doesn't cause a panic
+func TestConcurrentConfirmMissing(t *testing.T) {
+	found := []time.Time{
+		time.Unix(0, 0),
+		time.UnixMilli(1),
+		time.UnixMilli(2),
+	}
+	mr := &mockReader{resp: found}
+
+	output := &bytes.Buffer{}
+
+	wait := 30 * time.Millisecond
+	maxWait := 30 * time.Millisecond
+
+	c := NewComparator(output, wait, maxWait, 50*time.Hour, 15*time.Minute, 4*time.Hour, 4*time.Hour, 0, 1*time.Minute, 0, 0, 1, make(chan time.Time), make(chan time.Time), mr, false)
+
+	for _, t := range found {
+		c.missingEntries = append(c.missingEntries, &t)
+	}
+
+	for i := 0; i < 10; i++ {
+		go func() {
```

Comment:
better to add `sync.WaitGroup` to makesure all goroutines are completed by end of the test?

Reply:
good suggestion. adding

Reply:
Thanks for adding :)

1. can we make `wg.Done()` as `defer wg.Done()` at the beginning before `assert.NotPanic` is getting called? Because that way we can make sure `wg.Done()` is run even if that code panics.
2. Not familiar with `assert.Eventually`. But shouldn't just simple `wg.Wait()` after the for loop is sufficient here? Not sure if I miss anything there.

Reply:
I believe `assert.NotPanic` recovers from the panic and returns false. Adding the defer to make the code easier to reason about.

For 2, I would recommend keeping `Eventually` there. The expectation is that this test should return fairly quickly(within 1s) keeping a `wg.Done()` at the end will make the test suite time out(default 10m) if one of those goroutines hits a deadlock.